docs/Testing.md

```diff
@@ -1,13 +1,142 @@
 <h1 align="center">Fastify</h1>
 
 ## Testing
+
 Testing is one of the most important parts of developing an application. Fastify is very flexible when it comes to testing and is compatible with most testing frameworks (such as [Tap](https://www.npmjs.com/package/tap), which is used in the examples below).
 
-<a name="inject"></a>
-### Testing with http injection
+Let's `cd` into a fresh directory called 'testing-example' and type `npm init -y` in our terminal.
+
+run `npm install fastify && npm install tap --save-dev`
+
+### Separating concerns makes testing easy
+
+ First we're going to separate our application code from our server code:
+
+fastify.js
+
+```js 
+const Fastify = require('fastify')
+
+function buildFastify() {
+  const fastify = Fastify()
+  fastify.get('/', async (request, reply) => {
+    return { hello: 'world' }
+  })
+
+  return fastify
+}
+
+module.exports = buildFastify
+```
+
+index.js
+
+```js
+const server = require('./fastify')()
```

Comment:
If you allow to pass through some options to the `Fastify` factory, you can pass in a few logger options so that we can enable logging and pretty printing of said logs for development.

Reply:
Didn't know you could get pretty-print logging like that with IDs. I literally don't know how I would use this but I know it's probably amazing haha!